src/lib.rs

```diff
@@ -95,6 +96,34 @@ impl SocketType {
     }
 }
 
+/// Device types
+#[allow(non_camel_case_types)]
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum DeviceType {
+    STREAMER,
+    FORWARDER,
+    QUEUE,
+}
+
+impl DeviceType {
+    fn to_raw(self) -> c_int {
+        let raw = match self {
+            STREAMER => zmq_sys::ZMQ_STREAMER,
+            FORWARDER => zmq_sys::ZMQ_FORWARDER,
+            QUEUE => zmq_sys::ZMQ_QUEUE,
+        };
+        raw as c_int
+    }
```

Comment:
Could this return a result rather than a panic?

Also, could you add a newline before this line?